assets/training/distillation/components/data_generation/src/generate_data.py

```diff
@@ -260,52 +260,41 @@ def generate_synthetic_data(
         train_file_path (Path): Train JSONL file path
         validation_file_path (Path, optional): Validation JSONL file path. Defaults to None.
     """
-    def process_request(idx: str, enable_cot: bool, data: dict, url: str, endpoint_key: str) -> dict:
-        """Process a single request.
+
+    def process_system_prompt(message: dict):
+        """Update the system prompt depending on the task type and the flag enable_cot.
+
+        The original message unchanged if enable_cot is False or task type is conversation.
 
         Args:
-            idx (str): Row index in Input data.
-            enable_cot (bool): If CoT is enabled
-            data (dict): Payload dict
-            url (str): Endpoint URL
-            endpoint_key (str): key to authenticate endpoint request
+            message (dict): System message
 
         Returns:
-            dict: result dictionary
+        message: System message with updated content
         """
-        try:
-            response: Response = _invoke_endpoint(url=url, key=endpoint_key, data=data)
-            response_data = response.json()
-
-            # use jsonpath or regex to capture prediction result
-            prediction_result = (
-                None if response.status_code != 200
-                # response content should be structured as below for a successful vllm response
-                else response_data['choices'][0]["message"]["content"].strip()
-            )
+        if enable_cot and data_generation_task_type != DataGenerationTaskType.CONVERSATION:
```

Comment:
Will we add another parameter COD for Summarization task? What is the plan for it?

Reply:
+1, ideally we should use a single parameter to enable cot or cod, as they are mutually exclusive and involve similar set of operations discriminated by distillation task type

Reply:
@skasturi Hi Sasi, please can you comment on separate or unified flag for cod & cot ? I have a draft PR ready for summarization changes, so it would help to know what to use there.. [link](https://github.com/Azure/azureml-assets/pull/3278)

Reply:
Having two explicitly is okay. As part of validation, we can reset/ignore the one that is not relevant.